File: mobile/test/unit/design_context_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:ecole_platform/shared/ui/age_theme.dart';
import 'package:ecole_platform/shared/ui/design_context.dart';

void main() {
  group('school design context', () {
    test('defaults to formal for regular schools', () {
      expect(
        resolveSchoolDesignMode(role: 'ADM', schoolType: 'formal'),
        SchoolDesignMode.formal,
      );
    });

    test('uses settings override before school type', () {
      expect(
        resolveSchoolDesignMode(
          role: 'DIR',
          schoolType: 'formal',
          schoolSettings: {'design_mode': 'informal'},
        ),
        SchoolDesignMode.informal,
      );
    });

    test('forces informal for educator and micro routes', () {
      expect(
        resolveSchoolDesignMode(role: 'EDUCATOR', schoolType: 'formal'),
        SchoolDesignMode.informal,
      );
      expect(
        resolveSchoolDesignMode(route: '/micro-schools'),
        SchoolDesignMode.informal,
      );
    });
  });

  group('student age tier', () {
    test('resolves Moroccan class levels', () {
      expect(resolveAgeTier(niveau: 'CP'), AgeTier.primaire);
      expect(resolveAgeTier(niveau: '6eme'), AgeTier.college);
      expect(resolveAgeTier(niveau: '3AC'), AgeTier.college);
      expect(resolveAgeTier(niveau: 'Terminale'), AgeTier.college);
    });

    test('falls back to primaire when profile data is missing', () {
      expect(resolveAgeTier(), AgeTier.primaire);
    });
  });
}

File: web/tests/unit/shared/designContext.test.ts
import { describe, expect, it } from 'vitest';
import { resolveDesignContext } from '@/shared/ui/designContext';

describe('resolveDesignContext', () => {
  it('keeps formal as the default for non-student school contexts', () => {
    const context = resolveDesignContext({
      role: 'ADM',
      schoolType: 'formal',
      themeMode: 'light',
    });

    expect(context.designMode).toBe('formal');
    expect(context.schoolType).toBe('formal');
    expect(context.appliedTheme).toBe('light');
  });

  it('uses settings design_mode before school_type', () => {
    const context = resolveDesignContext({
      role: 'DIR',
      schoolType: 'formal',
      schoolSettings: { design_mode: 'informal' },
    });

    expect(context.designMode).toBe('informal');
  });

  it('forces informal for educator and micro routes', () => {
    expect(resolveDesignContext({ role: 'EDUCATOR', schoolType: 'formal' }).designMode).toBe(
      'informal',
    );
    expect(resolveDesignContext({ role: 'ADM', pathname: '/micro-schools' }).designMode).toBe(
      'informal',
    );
  });

  it('keeps student themes aligned with the school theme mode', () => {
    expect(resolveDesignContext({ role: 'STD', themeMode: 'light' }).appliedTheme).toBe('light');
    expect(resolveDesignContext({ role: 'STD', themeMode: 'dark' }).appliedTheme).toBe('dark');
  });
});
